### PR TITLE
[EBPF-616] gpu: Use event consumer for processs monitoring

### PR DESCRIPTION
**Pull request by gjulianm**
### What does this PR do?

This PR adds the process consumer introduced in https://github.com/DataDog/datadog-agent/pull/30559 to the GPU monitoring module. The consumer is created in the `EventMonitor` module and stored as a global variable in the GPU monitoring module.

### Describe how to test/QA your changes

Tested in e2e and KMT tests

### CODE REVIEW

**Review by val06 — APPROVED** [review]

File: cmd/system-probe/modules/eventmonitor.go [hunk 1]
```diff
@@ -91,5 +94,13 @@ func createEventMonitorModule(_ *sysconfigtypes.Config, deps module.FactoryDepen
 		}
 	}
 
+	gpucfg := gpuconfig.New()
+	if gpucfg.Enabled {
+		err := createGPUProcessEventConsumer(evm)
```
Comment (paulcacheux) [inline cmd/system-probe/modules/eventmonitor.go:99]: So I'm not 100% sure, and I still have a TODO list item to refactor this, but don't you need to call `evm.RegisterEventConsumer(` here, like other consumers ?

Comment (gjulianm) [reply cmd/system-probe/modules/eventmonitor.go:99]: That's included in the `createGPUProcessEventConsumer` function,  more specifically [here](https://github.com/DataDog/datadog-agent/blob/guillermo.julian/gpu-event-consumer/pkg/eventmonitor/consumers/process.go#L98) in the generic method for process-data consumers.

Comment (gjulianm) [reply cmd/system-probe/modules/eventmonitor.go:99]: That's a different consumer, for forwarding data to process-agent I think. This one is the one I introduced in [this PR](https://github.com/DataDog/datadog-agent/pull/30559) to have a generic consumer that can be used to replace the process monitor, which turned out to be only for USM use.

**Review by paulcacheux — APPROVED** [review]

**Pull request merged**